src/v/kafka/server/replicated_partition.cc

```diff
@@ -282,22 +282,26 @@ raft::replicate_stages replicated_partition::replicate(
 std::optional<model::offset> replicated_partition::get_leader_epoch_last_offset(
   kafka::leader_epoch epoch) const {
     const model::term_id term(epoch);
-    const auto first_local_term = _partition->get_term(
-      _partition->start_offset());
-    // found in local state
+    const auto first_local_offset = _partition->start_offset();
+    const auto first_local_term = _partition->get_term(first_local_offset);
+    // Found in local state
     if (term >= first_local_term) {
         auto last_offset = _partition->get_term_last_offset(term);
         if (last_offset) {
             return _translator->from_log_offset(*last_offset);
         }
     }
-
+    // The requested term falls below our earliest local segment.
+    // Check cloud storage for the highest offset in this term.
     if (
       _partition->is_remote_fetch_enabled()
       && _partition->cloud_data_available()) {
         return _partition->get_cloud_term_last_offset(term);
     }
-    return std::nullopt;
+    // Return the offset of the next-highest term, i.e. the start of the local
+    // log. This mirrors behavior in Kafka, see
+    // https://github.com/apache/kafka/pull/5678 for more details.
+    return _translator->from_log_offset(first_local_offset);
```

Comment:
this is great. how do we write a test that exercises this path.

Reply:
I updated one of the tests added when implementing KIP-320. Will look into testing more organically

Reply:
Opted to add a unit test, per @graphcareful's suggestion